v2/internal/store/insertionindex.go

```diff
@@ -237,3 +237,28 @@ func (ii *InsertionIndex) HasExactCID(c cid.Cid) bool {
 	ii.items.AscendGreaterOrEqual(entry, iter)
 	return found
 }
+
+func (ii *InsertionIndex) HasMultihash(mh multihash.Multihash) bool {
+	d, err := multihash.Decode(mh)
+	if err != nil {
+		panic(err)
```

Comment:
I can see some people not liking this; remember we're `panic()`-shy around here. It's pretty weird for it to fail at this step but it should be easy enough to just return `(bool, err)` from this function and then you can return the call in your `Has()`?